C/main_libFuzzer_extra_counters.c

```diff
@@ -0,0 +1,38 @@
+#include <stdio.h>
+#include <stddef.h>
+#include <stdlib.h>
+#include <stdint.h>
+#include <string.h>
+
+typedef long long GoInt64;
+typedef GoInt64 GoInt;
+typedef struct { void *data; GoInt len; GoInt cap; } GoSlice;
+extern void gofuzz_Init(void* coverTabPtr, uint64_t coverTabSize);
+extern void gofuzz_Run(GoSlice p0);
+
+
+#ifdef __linux__
+__attribute__((section("__libfuzzer_extra_counters")))
+#endif
+uint8_t Counters[65536];
+
+int LLVMFuzzerInitialize(int *_argc, char ***_argv) {
+    (void)_argc;
+    (void)_argv;
+
+    gofuzz_Init(Counters, sizeof(Counters));
+
+    return 0;
+}
+
+int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
+    uint8_t* datacopy = malloc(size);
```

Comment:
Why do we need to make a copy of the data rather then pass the data as is?

Reply:
Because ````const uint8_t data``` may not be passed to a function that does not treat the pointer as ```const```. But implementing ```LLVMFuzzerTestOneInput``` in Go and using ```SliceHeader``` solves this I think.